R/linelist_merge.R

```diff
@@ -0,0 +1,416 @@
+#must keep this pattern all one line or doesnt work
+
+    # filter variable cleaning dictionary specific to country file loaded
+    var_dict_country<- var_dict %>% filter(country==iso)
+    var_dict_country<- var_dict_country[var_dict_country$old_var_r %in% names(og_sheet),]
+
+    # list variables to drop that are non-template
+    drop_nontemplate_vars<- as.list(var_dict_country$old_var_r[var_dict_country$notes=="non_template" & !is.na(var_dict_country$notes)])
+
+    # drop variables from dataframe which are non-tmeplate
+    if(length(drop_nontemplate_vars)!=0){
+      og_sheet<-og_sheet[ , -which(names(og_sheet) %in% drop_nontemplate_vars)]
+    }
+
+    # if file requires no cleaning as matched WHO standard template do this loop
+    if(template==TRUE){
+      output_sheet<-og_sheet
+
+     #If file needs recoding of variables/merging of variables
+      }else if(template==FALSE){
+      # recode variables
+      recode_vars<-as.list(var_dict_country$old_var_r[grepl("Recode",var_dict_country$notes) & !is.na(var_dict_country$notes)])
+
+
+      if(length(recode_vars)!=0){
+      var_dict_country<- var_dict_country[var_dict_country$old_var_r %in% names(og_sheet),]
+      #from variable dictionary find which variables require recode
+      #these variables are usually patinfo_occus (proffession), comcond_exist (pre exsiting medical) and patsympt_other (other symptoms whihc have often been split inot separate variables)
+      variable_to<- sub("Recode to ", "",as.list(var_dict_country$notes[grepl("Recode",var_dict_country$notes) & !is.na(var_dict_country$notes)]))
+      what_to_match<-unique(variable_to)
+
+      recode_sheet<- og_sheet[ , which(names(og_sheet) %in% recode_vars)]
+
+      #remove following instances from variable names taht need recoding for the next step to leave what will become the recoded value
+      names(recode_sheet)<- gsub(x = names(recode_sheet), pattern = "COVID.|Co19.|comcond.|patsympt.|patinfo.", replacement = "", ignore.case = T)
+
+      #replace all instances of no with missing
+      recode_sheet<- data.frame(lapply(recode_sheet, function(x) {gsub("NAO|NON|NO|none|nil|^[nN]$|Know", NA, x, ignore.case = T, perl = T) }), stringsAsFactors=F)
+      #replace all instances of yes with 1
+      recode_sheet <- data.frame(lapply(recode_sheet, function(x) {gsub("SIM|OUI|YES|1|(?i)^si$|^[yY]$", 1, x, ignore.case = T, perl = T) }), stringsAsFactors=F)
+      #where all instances were 1 (yes) recode to the name of the variable for merge
+      recode_sheet <-  data.frame(sapply(names(recode_sheet), function(x)ifelse(recode_sheet[,x] == 1, x, recode_sheet[,x])), stringsAsFactors=F)
+      #renames variables with key of what to code to
+      names(recode_sheet)<-paste(names(recode_sheet),variable_to, sep= ".")
+
+      require(data.table)
```

Comment:
dont require packages in the function - add it to imports. Okay to use data.table here as is backend, but rethink if actually need to this because dependencies and breaking...

Reply:
recoded to not depend on data.table, will double checks works in all instances but seems ok